### DISCUSSION TITLE
List supported ESP32 boards

### DISCUSSION THREAD

**Discussion by zamojski** [emsesp/EMS-ESP32#839] [Q&A] [answered]
Can you confirm which ESP32 boards are compatible with the project? They differ in SoC, RAM size, interfaces, etc.

https://www.espressif.com/en/products/devkits

**Comment by MichaelDvP** [comment, marked as answer]
- Binaries are only provided for ESP32 soc. 
- Sourcecode is compatible with ESP32, ESP32S2, ESP32C3 and with a few lines addition ESP32S3,
- Flash minimum is 4M, more is supported
- Ethernet is only supported on ESP32 SOC.
- RAM on ESP32S2 is to low, only boards with extra PSRAM are working, maybe with some issues.
- PSRAM is supported for all SOCs.
- ESP32C3 codesize for RISC-V is a bit bigger, in future there is asymetric partitioning with loader/application needed. There is a test version.
- ESP32C2 is not supported and not planed. RAM is much to low.
- ESP32C6, not much known today, maybe in future.
- first USB flashing on S2, C3 works from platformio and esptool.exe, Esptool.py seems to have some isues with these socs, i did not get it working. Updating via web interface works.
- For operation only 2 pins for uart rx/tx are needed and can be mapped to (nearly) any pin, so most boards should work. Additional pins for LED, Dallas, Analog can also be mapped.

Tested boards as plugin in a premium gateway (esp8266 only with v2 software, others with v3.5.0b):
![ESP-mini_](https://user-images.githubusercontent.com/59284019/209464555-691aa510-1a69-45fa-9925-1d0699bf585f.jpg)

ESP32 have a large user base running on may systems. 
For C3, S2, S3 there is only a single test on one system, maybe with lots of unknown issues.
Best choose is a ESP32 dualcore with more than 4M Flash and maybe some PSRAM.
